dnn/utils/yolo_encoder.py

```diff
@@ -0,0 +1,138 @@
+import tensorflow as tf
+import numpy as np
+from absl.flags import FLAGS
+
+# maybe use tf.Tensor?
+YOLO_TINY_ANCHORS = np.array(
+    [[10, 14], 
+    [23, 27], 
+    [37, 58],
+    [81, 82], 
+    [135, 169],  
+    [344, 319]],
+    dtype = np.float32
+)
+
+YOLO_TINY_ANCHOR_MASKS = [[3,4,5], [1,2,3]]
+
+class yoloEncoder(tf.Module):
+    def __init__(self, size = 416, anchor = YOLO_TINY_ANCHORS, mask = YOLO_TINY_ANCHOR_MASKS):
+        '''
+        Args:
+            size: size of the image
+            anchor: anchor, array of shape(n, 2)
+            mask: anchor id that is masked for differnent dimension,
+                      For example, [[3,4,5], [1,2,3]] for yolov3-tiny
+        '''
+        super(yoloEncoder, self).__init__()
+        self.img_size = size
+        self.anchors_n2 = anchor / size
+        self.anchor_masks_n3 = mask
+
+
+    def transform_label_for_output(self, y_true_n6, grid_size, masks):
+        '''
+        Transform tensors that represent boxes into yolo_out
+
+        Args:
+            y_true: tensor of shape (boxes, (x1, y1, x2, y2, class, best_anchor))
+            grid_size: int, number of 'cut' per image. For yolov3-tiny, it's 13 and 26. 
+            masks: ordered anchor_ids of shape (1,n), e.g : (3,4,5)
+
+        '''
+        # mask out boxes that doesn't belong to this set of anchors
+        mask = tf.math.logical_and(
+            tf.math.greater_equal(y_true_n6[...,5], masks[0]), 
+            tf.math.less_equal(y_true_n6[...,5], masks[-1])) 
+        mask = tf.reshape(mask, (y_true_n6.shape[0],1))
+        mask = tf.broadcast_to(mask, y_true_n6.shape)
+        y_true_n6 = tf.boolean_mask(y_true_n6, mask)
```

Comment:
```suggestion
        mask_n6 = tf.broadcast_to(mask_n[:, tf.newaxis], y_true_n6.shape)
        y_true_n6 = tf.boolean_mask(y_true_n6, mask_n6)
```
Is this tested? I don't think `y_true_n6` is of shape `(n, 6)`. If I am not mistaken, `tf.boolean_mask` returns a 1D array

Reply:
It's tested before I did the wrapping up, I will double check

Reply:
Probably I messed during renaming

Reply:
Quick experiment
```python
mat = tf.constant([[0, 1], [2, 3]])
mask = tf.constant([[True, False], [True, False]])
# tf.boolean_mask(mat, mask) == [0, 2]
```
Here I think `mask` and `y_true_n6` are 2D arrays, the end result is 1D.